src/bosh-dns/dns/server/handlers/failover_recursor_pool.go

```diff
@@ -89,26 +111,48 @@ func newSmartFailoverRecursorPool(recursors []string, logger logger.Logger) Recu
 		preferredRecursorIndex: 0,
 		logger:                 logger,
 		logTag:                 logTag,
+		recursorRetrySettings:  recursorSettings,
 	}
 }
 
 func (q *serialFailoverRecursorPool) PerformStrategically(work func(string) error) error {
 	for _, r := range q.recursors {
-		if err := work(r); err == nil {
+		if err := performRetryLogic(work, r, q.recursorRetrySettings.retryCount, q.logTag, q.logger); err == nil {
 			return nil
 		}
 	}
 
 	return ErrNoRecursorResponse
 }
 
+func performRetryLogic(work func(string) error, recursor string, retryCount int, logTag string, log logger.Logger) (err error) {
```

Comment:
Naming is really hard and I'm not sure what fits better but `performRetryLogic` leaves the impression in me that the function does only retry logic. In fact the function does the initial DNS resolution and retry logic. What do you think about a name like `performWithRetryLogic` ?

Reply:
Fine for me